trxer.go

```diff
@@ -22,17 +22,20 @@ import "io"
 var UPDATE_INTERVAL = 5
 var PORT = 6666
 
-var BYTE_BUFFER_SIZE = 8096 * 8
+var DEF_BUFFER_SIZE = 8096 * 8
+var MIN_BUFFER_SIZE = 4096
```

Comment:
why an *artifical* minimum? If the user want to test 1 byte transfer, why the hell not?

Reply:
with write() we copy all data from application buffer to socket buffer right? then: when i check out for example the tcp_wmem variable I see 4096 is the minimum TCP send buffer space. So when the write() syscall comes in and the send buffer is empty...we instantly fill it up completely. with just one byte its a lot of overhead isnt it?

Reply:
Nope, not really. It is not *that* true, but true enough to explain the stuff:

`tcp_wmem` sets a minimum buffer size, this is the kernel internal buffer size allocated. But it is *no* limit to the syscall. The user can always send 1 byte of application data to the socket via write. `tcp_wmem` is just an kernel internal buffer tuning knob. And think about twice: you set an artificical limit in your application because you stumbled over `tcp_wmem`. Normally application programmer doent even know about `tcp_wmem`. So please remove the min limit, also the upper limit should be removed. The upper limit you encountered is probably just a *actual* limit, and will be removed in the future. But your application limit will stay forever. Another questions that arise: what about Windows, *BSD, ..? There are probably other limits somewhere? Hardcodeing a _current linux_ limits makes no sense again.